### ISSUE TITLE
OpenJ9 picks the wrong method from interfaces

### ISSUE THREAD

**Issue by andresluuk** [eclipse/openj9#2720]
The following code fails on openJ9 (I tried with 9 and 10), but passes on with the default java:
package pkg;

```
import java.lang.reflect.Method;

public class OpenJ9DefaultMethodTest {
  public static void main(String[] args) throws Exception {
    System.out.println("start");
    Method m = Interface.class.getDeclaredMethod("run");
    m.setAccessible(true);
    Target t = new Target();
    System.out.println("Is result 'Interface private method' from target ? " + m.invoke(t));
    System.out.println("done");
  }

  public interface Interface {

    private String run() {
      return "Interface private method";
    }
  }

  public interface Interface2 {

    default String run() {
      return "Interface2 default method";
    }
  }

  public static class Target implements Interface, Interface2 {
  }
}
```

Seems like openJ9 picks the wrong method from the implemented interfaces.

**Comment by fjeremic** [comment]
@pdbain-ibm @DanHeidinga FYI seems like your area of expertise.

**Comment by theresa-m** [comment]
Hi Andres, which versions of Java 9/10 are you using? I believe this was fixed with #2670

**Comment by andresluuk** [comment]
I downloaded my openJ9-s a few weeks ago. Definitely older then one week.
I now tried the latest nightly build of JDK10. My test did not fail there. So it was probably fixed in #2670

**Comment by DanHeidinga** [comment]
Closing based on previous comment.  Please reopen if this is still an issue.
